src/helper.cc

```diff
@@ -869,8 +874,35 @@ helper::handleKilledServer(HelperServerBase *srv, bool &needsNewServers)
 
         if (childs.needNew() > 0) {
             srv->flags.shutdown = true;
-            needsNewServers = true;
+            helperOpenServers(this);
```

Comment:
What about the `class statefulhelper`s which need this logic to call `helperOpenStatefulServer(this)` instead ?

Reply:
> Facilitate future reuse of ::helper class PR

This is now [PR1480](https://github.com/squid-cache/squid/pull/1480/files) .

Reply:
> Wait for a prerequisite minimal PR that makes (the equivalent of) helperOpenServers() virtual.

Done in recent official commit bd71920. That virtual method was named openSessions(). This PR has been updated to use that method, fixing this bug.